saleor/core/jwt.py

```diff
@@ -110,10 +110,10 @@ def get_user_from_access_token(token: str) -> Optional[User]:
     payload = jwt_decode(token)
     if payload["type"] not in [JWT_ACCESS_TYPE, JWT_THIRDPARTY_ACCESS_TYPE]:
         return None
-    permission_limits = payload.get(PERMISSION_LIMITS_FIELD, None)
+    permissions = payload.get(PERMISSIONS_FIELD, None)
     user = get_user_from_payload(payload)
-    if permission_limits is not None:
-        user.permission_limits = permission_limits  # type: ignore
+    if user and permissions is not None:
+        user.effective_permissions = get_permissions_from_names(permissions)
```

Comment:
Shouldn't we cross-check those with the database permissions? Can a token grant me permission I did not previously hold?

Reply:
Unless the secret is known or was leaked (not impossible!), the HMAC should ensure nothing was altered/injected. What we can't ensure, is that permissions are actually up to date.

Reply:
@patrys Yes, I am adding it now :)